src/control.f90

```diff
@@ -824,6 +826,23 @@ subroutine md_run (fixed_potential, vary_mu, total_energy)
 
     do iter = i_first, i_last ! Main MD loop
        mdl%step = iter
+       
+       ! At a given time step, if the difference between T_ext and the final temperature is larger than 
+       ! the temperature step (with 5% margin), then update T_ext 
+       ! Temperature evolves linearly from temp_ion to temp_ion_end over MDn_steps steps
+       ! Stop when target temperature has been reached
+       if  (abs(mdl%T_ext - temp_ion_end) > 0.95*abs((temp_ion_end-temp_ion)/dble(MDn_steps))) then
+         mdl%T_ext = temp_ion+(dble(iter-1)/dble(MDn_steps))*(temp_ion_end-temp_ion)
+         write(io_lun,fmt='(6x, "Thermostat temperature at step ", i5, ": ", f9.1, " K")') iter, mdl%T_ext 
+         ! Update target ke for SVR
+         thermo%ke_target = half*md_ndof_ions*fac_Kelvin2Hartree*mdl%T_ext
+         write(io_lun,fmt='(6x, "kee target is now" , f8.3)') thermo%ke_target
+         
+       else if (abs(temp_ion_end-temp_ion) > 0.0001) then
+         write(io_lun,fmt='(6x, "Target temperature (", f9.1," K) has been reached. Stopping..")') mdl%T_ext 
```

Comment:
This needs to be only on `ionode` (or `if(myid==0)` and only when `iprint` is high enough. Please also make the 0.0001 parameter adjustable

Reply:
I realize that setting this small number was unnecessary. Simply checking that `temp_ion/=temp_ion_end` should be sufficient to trigger this exit command.

```
       if [...]
         [...]
       else if (temp_ion_end /= temp_ion) then ! Ensure that temp_ion_end was different from temp_ion
         if (inode == ionode) then
           write(io_lun,fmt='(6x, "Target temperature (", f9.1," K) has been reached. Stopping..")') mdl%T_ext
         end if
		 exit
       end if
```

Reply:
It's never safe to use `/=` to compare two real numbers.  We have a parameter `RD_ERR` in `numbers` module that would work here:

```
else if(abs(temp_ion_end - temp_ion)>RD_ERR) then...
```

would be much better.

Reply:
I see! Then I will replace the condition containing hard-coded 0.0001 with RD_ERR.